api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageQueueDAO.java

```diff
@@ -24,26 +25,30 @@
 public class MessageQueueDAO implements MessageQueueClient {
 
     private final MessageRepository messageRepository;
-
     private final ObjectMapper objectMapper;
 
     @Override
     public void sendMessage(MessagePayload messagePayload, MessageType messageType) {
+        sendMessageWithDelay(messagePayload, messageType, Period.ZERO);
+    }
+
+    @Override
+    public void sendMessageWithDelay(MessagePayload messagePayload, MessageType messageType, Period messageDelay) {
         try {
-            messageRepository.save(buildMessage(messagePayload, messageType));
+            messageRepository.save(buildMessage(messagePayload, messageType, messageDelay));
         } catch (JsonProcessingException e) {
             throw new MessageProcessingException("Unable to create JSON payload from object", e);
         }
     }
 
-    private Message buildMessage(MessagePayload messagePayload, MessageType messageType) throws JsonProcessingException {
+    private Message buildMessage(MessagePayload messagePayload, MessageType messageType, Period messageDelay) throws JsonProcessingException {
         String json = objectMapper.writeValueAsString(messagePayload);
+        LocalDateTime messageTimestamp = LocalDateTime.now().plus(messageDelay);
         return Message.builder()
                 .messagePayload(json)
                 .messageType(messageType)
-                .messageTimestamp(LocalDateTime.now())
+                .messageTimestamp(messageTimestamp)
```

Comment:
What did we agree for a rename of this variable, was it `processTimestamp`?

Reply:
I think it was 'processAfterTimestamp', but that we talked about keeping the column name as 'message_timestamp'.